backend/keepalived/keepalived.go

```diff
@@ -473,32 +479,30 @@ func (k *Keepalived) dead(key string, prev liveness.State, leader bool) bool {
 	}
 
 	ctx := store.NamespaceContext(k.ctx, namespace)
+	meta := corev2.NewObjectMeta(name, namespace)
+	entityConfig := &corev3.EntityConfig{Metadata: &meta}
 
-	entity, err := k.store.GetEntityByName(ctx, name)
+	req := storev2.NewResourceRequestFromResource(ctx, entityConfig)
+	wrapper, err := k.storev2.Get(req)
 	if err != nil {
-		lager.WithError(err).Error("error while reading entity")
+		if _, ok := err.(*store.ErrNotFound); ok {
+			// The entity has been deleted, there is no longer a need to track
+			// keepalives for it.
+			lager.Debug("nil entity")
+			return true
+		}
+		lager.WithError(err).Error("error while reading entity_config")
 		return false
 	}
 
-	if entity == nil {
-		// The entity has been deleted, there is no longer a need to
-		// track keepalives for it.
-		lager.Debug("nil entity")
-		return true
+	resource, err := wrapper.Unwrap()
```

Comment:
When you know the type of thing you are trying to unwrap, you can use `UnwrapInto`. This would save a few lines of code.

```
var entity corev3.EntityConfig
if err := wrapper.UnwrapInto(&entity); err != nil {
  return err
}
```